dashboard/app/models/ability.rb

```diff
@@ -105,6 +105,31 @@ def initialize(user)
         can?(:manage, section) || user.sections_as_student.include?(section)
       end
 
+      can :view_as_user, ScriptLevel do |script_level, user_to_assume|
+        can_view_as_user = false
+        can_view_as_user = true if user_to_assume.student_of?(user) ||
+          user.project_validator?
+
+        # Only allow a student to view another student's project
+        # only on levels where we have our peer review feature.
+        # For now, that's only Javalab.
+        if script_level.oldest_active_level.is_a?(Javalab)
+          reviewable_project = ReviewableProject.find_by(
+            user_id: user_to_assume.id,
+            script_id: script_level.script_id,
+            level_id: script_level.oldest_active_level&.id
+          )
+
+          if reviewable_project &&
+            user != user_to_assume &&
+            (user.sections_as_student & user_to_assume.sections_as_student).any?
```

Comment:
nit: seems like often need to check if two students are in the same section in this manner, is there a utility/shared class we can move this check into?

Reply:
Yeah fair point, this looks like it effectively does the same thing:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L1851

Going to leave as-is for now, but can update if this comes around again.